conda/cli/main_config.py

```diff
@@ -809,3 +833,34 @@ def execute_config(args, parser):
         from .common import stdout_json_success
 
         stdout_json_success(rc_path=rc_path, warnings=json_warnings, get=json_get)
+
+
+def migrate_pkgs(context, config: dict, json_warnings: list[str]):
+    """Migrate the pkgs/ directory from the root prefix to the user data directory.
+
+    Parameters
+    ----------
+    context : Context
+        Current execution context
+    config : dict
+        Configuration dict read from `.condarc`
+    json_warnings : list[str]
+        Warnings to be emitted
+    """
+    from .. import CondaError
+    from ..base.context import user_data_pkgs
+    from ..common.path.directories import hardlink_dir_contents
+
+    key = "pkgs_dirs"
+    if key in config:
+        raise CondaError(
+            f"The conda configuration for {key} has explicitly set and"
+            "cannot be migrated."
+        )
+
+    # There will be two copies of pkgs now; the legacy directory will
+    # be cleaned up with the next `conda clean`
+    hardlink_dir_contents(
```

Comment:
Hardlinks are not supported by all filesystems and cannot be made across filesystems. A fallback to copy is needed in these cases.

Additionally, the `always_copy` setting should be considered. If this is `true` I think a copy should be made. This is imprecise as the setting controls what happens when files are created in a prefix which is not the same as what is done here.

Reply:
Thanks for the feedback - I've added copying as a fallback. Also, the user can now specify `always_copy` to bypass hardlinking entirely.